crates/accelerate/src/quantum_circuit/intern_context.rs

```diff
@@ -0,0 +1,144 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2023
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use hashbrown::HashMap;
+use pyo3::prelude::*;
+use std::collections::hash_map::DefaultHasher;
+use std::collections::VecDeque;
+use std::hash::{Hash, Hasher};
+use std::mem::take;
+
+macro_rules! println {
+    ($($rest:tt)*) => {
+        #[cfg(debug_interner)]
+        std::println!($($rest)*)
+    }
+}
+
+#[cfg(debug_interner)]
+fn unique_id() -> u64 {
+    use std::sync::atomic::AtomicU64;
+    use std::sync::atomic::Ordering::SeqCst;
+    static COUNTER: AtomicU64 = AtomicU64::new(0);
+    COUNTER.fetch_add(1, SeqCst)
+}
```

Comment:
This is just temporary, right?  Even with the feature guard, I think it's probably best not to let it merge.

Reply:
Yeah, it's mostly just here in case someone wanted to build this demo branch to see for themselves that the slot allocation stuff is working (...since I don't have unit tests for it yet 😛).

```
RUSTFLAGS="--cfg debug_interner" python setup.py build_rust --release --inplace
```

Reply:
Removed in 8447c02.